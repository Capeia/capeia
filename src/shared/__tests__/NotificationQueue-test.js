// @flow
/* eslint-env jest, jasmine */

import React from 'react'
import renderer from 'react-test-renderer'

describe('<NotificationQueue>', () => {
  it('renders correctly', () => {
    jest.deepUnmock('../NotificationQueue')
    const Component = require('../NotificationQueue').default
    let notify
    const instance = renderer.create(
      /* eslint-disable react/jsx-no-bind */
      <Component register={(fn) => { notify = fn }} max={3} />
      /* eslint-enable react/jsx-no-bind */
    )
    let tree = instance.toJSON()
    expect(tree).toMatchSnapshot()

    const notifications = [
      {content: 'hello world'},
      {content: <span>foo</span>, type: 'error'},
      {content: <div>baz</div>, type: 'info'},
      {content: 'bar', type: 'success'}
    ]

    notifications.forEach(n => {
      notify(n.content, n.type || undefined)
      tree = instance.toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
