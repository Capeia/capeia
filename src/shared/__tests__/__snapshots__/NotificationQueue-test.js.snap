// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotificationQueue> renders correctly 1`] = `<div />`;

exports[`<NotificationQueue> renders correctly 2`] = `
<div>
  <div
    className="notification info"
  >
    hello world
  </div>
</div>
`;

exports[`<NotificationQueue> renders correctly 3`] = `
<div>
  <div
    className="notification error"
  >
    <span>
      foo
    </span>
  </div>
  <div
    className="notification info"
  >
    hello world
  </div>
</div>
`;

exports[`<NotificationQueue> renders correctly 4`] = `
<div>
  <div
    className="notification info"
  >
    <div>
      baz
    </div>
  </div>
  <div
    className="notification error"
  >
    <span>
      foo
    </span>
  </div>
  <div
    className="notification info"
  >
    hello world
  </div>
</div>
`;

exports[`<NotificationQueue> renders correctly 5`] = `
<div>
  <div
    className="notification success"
  >
    bar
  </div>
  <div
    className="notification info"
  >
    <div>
      baz
    </div>
  </div>
  <div
    className="notification error"
  >
    <span>
      foo
    </span>
  </div>
</div>
`;
